system/lib/wasmfs/wasmfs.cpp

```diff
@@ -70,14 +70,18 @@ WasmFS::~WasmFS() {
   rootDirectory->locked().setParent(nullptr);
 }
 
-std::shared_ptr<Directory> WasmFS::initRootDirectory() {
-
+// Special backends that want to install themselves as the root use this hook.
+// Otherwise, we use the default backends.
+__attribute__((weak)) extern backend_t wasmfs_create_root_dir(void) {
 #ifdef WASMFS_CASE_INSENSITIVE
-  auto rootBackend =
-    createIgnoreCaseBackend([]() { return createMemoryBackend(); });
+  return createIgnoreCaseBackend([]() { return createMemoryBackend(); });
```

Comment:
It would be nice to handle the `CASE_INSENSITIVE` option the same way, perhaps as a follow-up?

Reply:
Hmm, I was thinking about that. Do we want it to compose, that is, whatever the root backend is, in case-insensitive mode we'd wrap it? But I'm not sure if wrapping node makes sense (the underlying FS has its own case sensitivity).

Reply:
Yeah, I don't think it needs to compose with NODERAWFS.